jetty-ee10/jetty-ee10-webapp/src/test/java/org/eclipse/jetty/ee10/webapp/StandardDescriptorProcessorTest.java

```diff
@@ -53,6 +59,104 @@ public void afterEach() throws Exception
+        defaultServlet.setInitParameter("etags", "true");
+        defaultServlet.setInitParameter("useFileMappedBuffer", "false");
+        defaultServlet.setInitOrder(2);
+        defaultServlet.setRunAsRole("foo");
+        wac.getServletHandler().addServletWithMapping(defaultServlet, "/");
+        wac.start();
+
+        ServletHolder[] holders = wac.getServletHandler().getServlets();
+        ServletHolder holder = null;
+        for (ServletHolder h:holders)
+        {
+            if ("default".equals(h.getName()))
+            {
+                assertEquals(null, holder);
+                holder = h;
+            }
+        }
+        assertNotNull(holder);
+        assertEquals("false", holder.getInitParameter("acceptRanges"));
+        assertEquals("false", holder.getInitParameter("dirAllowed"));
+        assertEquals("true", holder.getInitParameter("welcomeServlets"));
+        assertEquals("true", holder.getInitParameter("redirectWelcome"));
+        assertEquals("10", holder.getInitParameter("maxCacheSize"));
+        assertEquals("1", holder.getInitParameter("maxCachedFileSize"));
+        assertEquals("10", holder.getInitParameter("maxCacheFiles"));
+        assertEquals("true", holder.getInitParameter("etags"));
+        assertEquals("false", holder.getInitParameter("useFileMappedBuffer"));
+        assertEquals(2, holder.getInitOrder());
+        assertEquals("foo", holder.getRunAsRole());
+    }
+
+    @Test
+    public void testDuplicateServletMappingsFromJettyApi(WorkDir workDir) throws Exception
+    {
+        Path docroot = workDir.getEmptyPathDir();
+        WebAppContext wac = new WebAppContext();
+        wac.setServer(_server);
+        wac.setBaseResourceAsPath(docroot);
+        wac.setThrowUnavailableOnStartupException(true);
+        //add a mapping that will conflict with the webdefault-ee10.xml mapping
+        //for the default servlet
+        ServletHolder defaultServlet = new ServletHolder(DefaultServlet.class);
+        defaultServlet.setName("noname");
+        wac.getServletHandler().addServletWithMapping(defaultServlet, "/");
+        try (StacklessLogging ignored = new StacklessLogging(WebAppContext.class))
+        {
+            assertThrows(InvocationTargetException.class, () -> wac.start());
+        }
+    }
+
+    @Test
+    public void testDuplicateServletMappingsFromDescriptors(WorkDir workDir) throws Exception
+    {
+        //Test that the DefaultServlet mapping from webdefault-ee10.xml can be overridden in web.xml
+        Path docroot = workDir.getEmptyPathDir();
+        File webXml = MavenTestingUtils.getTestResourceFile("web-redefine-mapping.xml");
+        WebAppContext wac = new WebAppContext();
+        wac.setServer(_server);
+        wac.setBaseResourceAsPath(docroot);
+        wac.setDescriptor(webXml.toURI().toURL().toString());
+        wac.start();
+        assertEquals("other", wac.getServletHandler().getServletMapping("/").getServletName());
```

Comment:
This test doesn't seem to assert that there is only 1 mapping for `/`.
Just that the first mapping for `/` is the expected servlet-name.